docs/TypeScript.md

```diff
@@ -125,6 +125,32 @@ The type system heavily relies on generic properties to provide the most accurat
 
 🎉 Good work, now you can define interfaces for each route and have strictly typed request and reply instances. Other parts of the Fastify type system rely on generic properties. Make sure to reference the detailed type system documentation below to learn more about what is available.
 
+### Type checking responses based on the status code
+
+If you would like to be strict about the responses that are returned to the caller of a route. You can use the RepliesType when creating a route this key allows a type to be specified for any status code.  When `.sendWithStatus` is called on the reply object, both the status code and the payload data is type scheduled to match.
+
+   ```typescript
+   server.get<{
+    RepliesTyped: {
+      200: {
+        username: string,
+      },
+      404: {
+        resultMessage: 'Not Found'
+        retryable: boolean,
+      }
```

Comment:
Can you add the schema block with the actual JSON schema for the output?

Something like the following for each status code:

```js
{
  schema: {
    response: {
      200: {
        type: 'object',
        properties: {
          value: { type: 'string' },
          otherValue: { type: 'boolean' }
        }
      }
    }
  }
}
```

A full example using one of the libraries you aim to support would also be helpful.